components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+                const colGroup = this.querySelector('colgroup');
+                const thead = this.querySelector('thead');
+                const tbody = this.querySelector('tbody');
+
+                // remove previous values
+                clearElement(colGroup);
+                clearElement(thead);
+                clearElement(tbody);
+
+                // remove any errors on this form
+                this.form.removeAttribute('data-error');
+                const formError = document.querySelector('ca-form [data-error]');
+                if (formError) {
+                    formError.removeAttribute('data-error');
+                }
+
+                if (headers.length > 0) {
+
+                    const tr = createElement(null, 'tr');
+
+                    // insert table headers
+                    for (let i = 0, l = headers.length; i < l; i++) {
+
+                        const label = headers[i].replace('*', '');
+                        const key = toCamelCase(label.trim());
+
+                        // insert selection col/header cell
+                        createElement(colGroup, 'col', { 'data-key': key });
+
+                        // insert header cell into header row
+                        const th = createElement(tr, 'th', { 'data-key': key });
+
+                        // insert value within a span to allow us to alter its value via CSS (hide span, insert content before etc)
+                        createElement(th, 'span', null, label);
+                    }
+
+                    thead.appendChild(tr);
+                }
+
+                data.forEach((item, index) => {
+
+                    const tr = createElement(null, 'tr');
+
+                    for (let i = 0, l = headers.length; i < l; i++) {
+
+                        const key = headers[i].replace('*', '');
+                        const value = item[key] || item[i] || '';
+
+                        createElement(tr, 'td', { title: value }, value);
+                    }
+
+                    tbody.appendChild(tr);
+                });
+            }
+        }
+
+        /**
+         * if we have first row values, bind with add them as an example row (to be removed during import)
+         * @returns {array} example row.
+         */
+        buildExampleRow() {
+
+            if (this.firstRow.length > 0) {
+
+                const exampleRow = {};
+                const len = Math.min(this.headers.length, this.firstRow.length);
+
+                for (let i = 0; i < len; i++) {
+                    exampleRow[this.headers[i]] = this.firstRow[i] || '';
+                }
+
+                return exampleRow;
+            }
+
+            return null;
+        }
+
+        downloadTemplateHandler(e = event) {
+            const el = e.target || e.srcElement;
+
+            switch (this.type) {
+
+                case 'csv': el.href = `data:text/plain;charset=utf-8,${encodeURIComponent(this.headers)}`; break;
```

Comment:
Noted. But let's put in a lint rule for this please so it's a rule.

We seem to use single if (cond) return; but case: cond: statement: break; isn't too different.